byway/src/main/webapp/routepage.js

```diff
@@ -292,5 +292,35 @@ function createRecButton(rec){
   return recBtn;
 }
 
-/* exported initMap, generateRoute, placesService, map */
-/* global google, findPlace, getTripKeyFromUrl, configureTripKeyForPath */
\ No newline at end of file
+
+/** Creates a URL link to Google Maps based on the start/end point and route
+ *  @returns {String} routeLink url containing query params for the user’s route
+ */
+function generateRouteLink(){
+    let routeLink = "https://www.google.com/maps/dir/?api=1&travelmode=driving" 
+    routeLink += "&origin=" + start.name.split(' ').join('%20') + "&destination=" + 
+      end.name.split(' ').join('%20') + "&waypoints=";
+    for(let place of route){
+      routeLink += place.name.split(' ').join('%20') + "|";
+    }
+   return routeLink;
+}
+
+/** Opens a new window of an email client with prefilled to, subject, and body field 
+ *  containing a Google Maps link of the user's route 
+ */
+function sendEmail(){
+  fetch('/api/email')
```

Comment:
Why not just let the user fill out an email address themselves? What if they want to mail it to someone else? Seems easier to have a box where they can enter an email address and then generate the email rather than getting it from the user data on the server.

Reply:
I added a field instead.